File: src/main/javabackup/de/thatsich/intellie/common/registries/TileEntityRegistry.java
package de.thatsich.intellie.common.registries;

import cpw.mods.fml.common.registry.GameRegistry;
import de.thatsich.intellie.common.module.tileentity.ITileEntity;
import de.thatsich.intellie.common.util.logging.ILog;
import net.minecraft.tileentity.TileEntity;
import net.minecraftforge.common.config.Configuration;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.Collection;
import java.util.LinkedList;

/**
 Registry for TileEntities
 */
@Singleton
public class TileEntityRegistry implements IRegistry<ITileEntity>
{
	private final Collection<ITileEntity> tileEntities;
	private final ILog log;

	@Inject
	TileEntityRegistry( final ILog log )
	{
		this.tileEntities = new LinkedList<>();
		this.log = log;
	}

	/**
	 Loads each config through their own responsibility

	 @param config Configuration
	 */
	public void loadConfig( final Configuration config )
	{
		for( ITileEntity tileEntity : this.tileEntities )
		{
			tileEntity.register( config );
		}
	}

	@Override
	public void add( final ITileEntity tileEntity )
	{
		this.log.info( "Added Tile Entity: %s", tileEntity );
		this.tileEntities.add( tileEntity );
	}

	@Override
	public void register()
	{
		for( ITileEntity tileEntity : this.tileEntities )
		{
			final TileEntity te = (TileEntity) tileEntity;
			final Class<? extends TileEntity> tileEntityClass = te.getClass();
			final String tileEntityKey = tileEntity.getTileEntityID();

			GameRegistry.registerTileEntity( tileEntityClass, tileEntityKey );
		}
	}
}

File: src/main/scala/de/thatsich/minecraft/core/registries/TileEntityRegistry.scala
package de.thatsich.minecraft.core.registries

import net.minecraft.tileentity.TileEntity
import cpw.mods.fml.common.registry.GameRegistry
import de.thatsich.minecraft.core.module.tileentities.ITileEntity
import de.thatsich.minecraft.core.log.ILog

/**
 *
 *
 * @author thatsIch
 * @since 07.04.2014.
 */
class TileEntityRegistry(log: ILog)
	extends ARegistry[ ITileEntity ](log)
{
	def register()
	{
		for( tileEntity <- this.set )
		{
			val te: TileEntity = tileEntity.asInstanceOf[ TileEntity ]
			val tileEntityClass = te.getClass
			val tileEntityKey = tileEntity.getTileEntityID
			GameRegistry.registerTileEntity(tileEntityClass, tileEntityKey)
		}
	}
}
